javascript/ql/lib/semmle/javascript/frameworks/Testing.qll

```diff
@@ -39,12 +39,14 @@ class BDDTest extends Test, @call_expr {
 }
 
 /**
- * Gets the test file for `f` with stem extension `stemExt`.
- * That is, a file named file named `<base>.<stemExt>.<ext>` in the
+ * Gets the test file for `f` with stem extension `stemExt`, where `stemExt` is "test" or "spec".
+ * That is, a file named `<base>.<stemExt>.<ext>` in the
  * same directory as `f` which is named `<base>.<ext>`.
  */
-bindingset[stemExt]
+pragma[noinline]
```

Comment:
Did you see the magic optimisation attempting to push context into this predicate? If so, make this `nomagic`.

Reply:
I did not, but I also didn't look for it.  
I just saw that the predicate disappeared from the clause timing report.  

The `pragma[inline]` wasn't added because I saw a bad join order, it was just added because of my gut feeling about how the predicate should be evaluated.